### DISCUSSION TITLE
Rename to @types/ts-reset

### DISCUSSION THREAD

**Discussion by kvnwolf** [mattpocock/ts-reset#196] [Ideas]
So far, I've been able to abstract this into my personal [shareable tsconfig.json](https://github.com/kevinwolfcr/devtools/blob/main/configs/tsconfig/tsconfig.json#L16) by adding `@total-typescript/ts-reset` to the `types` array of `tsconfig.json`.

However, by doing that, TypeScript does not automatically load `@types/*` packages, which forces me to re-declare the `types` array and manually include all the typings I want.

**Comment by mattpocock** [comment]
Yes, this is not a recommended approach - hence why the installation instructions are different.

**Reply by kvnwolf** [reply]
Understood. Why would you say it is not a recommended approach?

**Reply by mattpocock** [reply]
Because of what you pointed out, that "types" is exclusive.

**Reply by kvnwolf** [reply]
I agree, and I understand your point. One possible solution could be to create a new package named @types/ts-reset. If you’d rather not handle this yourself or rename the current library, would you be okay with me creating and publishing it to npm? Naturally, I’d give proper credit to the original library and clearly explain what it solves.

**Reply by mattpocock** [reply]
I'd prefer if it was named something else, as I'm not sure if I would be comfortable with an approach like that. I think that it's too implicit to just have an npm package change global types. An obvious exception is @types/node, but I think that @types/node and @types/ts-reset are quite different.

**Reply by kvnwolf** [reply]
As I see it, and the reason why I really like this library, its because it solves most of the caveats TypeScript has. So I'll maybe pick another name and just reference `@total-typescript/ts-reset`. Maybe something like `@types/ts-preflight` or `@types/ts-fix`.

**Reply by mattpocock** [reply]
Sure, though - as I said - I don't recommend that this be done.
